crates/ruma-signatures/src/keys.rs

```diff
@@ -181,7 +181,7 @@ pub type PublicKeyMap = BTreeMap<String, PublicKeySet>;
 /// A set of public keys for a single homeserver.
 ///
 /// This is represented as a map from key ID to base64-encoded signature.
-pub type PublicKeySet = BTreeMap<String, Base64>;
+pub type PublicKeySet = BTreeMap<OwnedServerSigningKeyId, Base64>;
```

Comment:
Related to this change, is there any value for `PublicKeyMap` to use `OwnedServerName` for the keys?

Reply:
I think it could be good, would make working with this function much more clear, and also ensure that consumers of the function are passing in the correct type.

Reply:
Should I do that in this pull request, or a new one?

Reply:
You can do that separately, it will make this PR a bit smaller.

Reply:
Looking more into it, both `sign_json` and `verify_json` could be used on the client side (to sign devices or cross-signing signatures), so they shouldn't use server signatures specific types.

Reply:
Seems like this isn't as big of a deal as I thought for us anyways, and if there is a usecase outside of servers, then I think it would be best to close this then.